sdk/storage/azure-storage-blob-changefeed/src/main/java/com/azure/storage/blob/changefeed/implementation/models/BlobChangefeedEvent.java

```diff
@@ -0,0 +1,212 @@
+        AvroSchema.checkType("r", record, Map.class);
+        Map<?, ?> r = (Map<?, ?>) record;
+
+        if (!r.get(AvroConstants.RECORD).equals("BlobChangeEvent")) {
+            throw new IllegalArgumentException("Not a valid BlobChangefeedEvent.");
+        }
+
+        Object topic = r.get("topic");
+        Object subject = r.get("subject");
+        Object eventType = r.get("eventType");
+        Object eventTime = r.get("eventTime");
+        Object id = r.get("id");
+        Object data = r.get("data");
+        Object dataVersion = r.get("dataVersion");
+        Object metadataVersion = r.get("metadataVersion");
+
+        return new BlobChangefeedEvent(
+            nullOrString("topic", topic),
+            nullOrString("subject", subject),
+            isNull(eventType) ? null
+                : BlobChangefeedEventType.fromString(BlobChangefeedEvent.nullOrString("eventType", eventType)),
+            isNull(eventTime) ? null
+                : OffsetDateTime.parse(Objects.requireNonNull(nullOrString("eventTime", eventTime))),
+            nullOrString("id", id),
+            isNull(data) ? null : BlobChangefeedEventData.fromRecord(data),
+            nullOrLong("dataVersion", dataVersion),
+            nullOrString("metadataVersion", metadataVersion)
+        );
+    }
+
+    /**
+     * Determines whether or not the object is null in the Avro sense.
+     */
+    static boolean isNull(Object o) {
+        return o == null || o instanceof AvroNullSchema.Null;
+    }
+
+    /**
+     * Returns either null or a String.
+     */
+    static String nullOrString(String name, Object o) {
+        if (isNull(o)) {
+            return null;
+        }
+        AvroSchema.checkType(name, o, String.class);
+        return (String) o;
+    }
+
+    /**
+     * Returns either null or a Long.
+     */
+    static Long nullOrLong(String name, Object o) {
+        if (isNull(o)) {
+            return null;
+        }
+        AvroSchema.checkType(name, o, Long.class);
+        return (Long) o;
+    }
+
+    /**
+     * Returns either null or a Boolean.
+     */
+    static boolean nullOrBoolean(String name, Object o) {
+        if (isNull(o)) {
+            return false;
+        }
+        AvroSchema.checkType(name, o, Boolean.class);
+        return (boolean) o;
+    }
+
+    /**
+     * @return the topic.
+     */
+    public String getTopic() {
+        return topic;
+    }
+
+    /**
+     * @return the subject.
+     */
+    public String getSubject() {
+        return subject;
+    }
+
+    /**
+     * @return {@link BlobChangefeedEventType}
+     */
+    public BlobChangefeedEventType getEventType() {
+        return eventType;
+    }
+
+    /**
+     * @return The {@link OffsetDateTime event time}.
+     */
+    public OffsetDateTime getEventTime() {
+        return eventTime;
+    }
+
+    /**
+     * @return the identifer.
+     */
+    public String getId() {
+        return id;
+    }
```

Comment:
we should mark all methods coming from interface with `@Override` (so that compiler can catch changes).
See https://stackoverflow.com/questions/212614/should-we-override-an-interfaces-method-implementation


We can also use `{@inheritDoc}` to not repeat javadocs.

This applies to other places so I won't repeat this there.

Reply:
Another fun note is Javadoc will walk the hierarchy tree if a method is annotated with `@Override` and doesn't have Javadocs to use parent Javadocs (won't use them if the class is `Object`).

Reply:
@alzimmermsft There's a rule in checkstyle to catch Override thing - could you guys enable it ?

Reply:
Should be easy to add, I'll file an issue for this as it may trigger a lot of new exceptions.

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/12265